hash.c

```diff
@@ -6736,9 +6912,24 @@ env_update(VALUE env, VALUE hash)
  *
  *  You can set the default proc to +nil+, which restores control to the default value:
  *
+ *    h.delete(:nosuch)
  *    h.default_proc = nil
  *    h.default = false
  *    h[:nosuch] # => false
+ *
+ *  === Chaining \Method Calls
+ *
+ *  Some \Hash instance methods return <tt>self</tt>.
+ *  For those methods, you can "chain" method calls
+ *  by following one method call with another.
+ *
+ *  This example chains methods #merge! and #compact!:
+ *
+ *  - <tt>merge!</tt> merges another Hash into <tt>h</tt>.
```

Comment:
I think you need to check this. Sometimes these methods can return `nil`.

Reply:
Fixed.